crates/next-api/src/project.rs

```diff
@@ -553,6 +557,9 @@ impl Project {
         let node_root = self.node_root();
         let next_mode = self.next_mode().await?;
 
+        // NOTE(LichuAcu): We don't call .module_id_strategy() here because, for App, when calling
```

Comment:
The execution_context doesn't need a global module ids. It's only used for execution during build.
As we need the execution_context for build modules, you can't have access to the global module graph.
Just give it the dev module ids always.